src/main/kotlin/org/rust/cargo/toolchain/RsToolchainBase.kt

```diff
@@ -139,9 +141,41 @@ abstract class RsToolchainBase(val location: Path) {
 
             return RsToolchainFlavor.getApplicableFlavors()
                 .asSequence()
-                .flatMap { it.suggestHomePaths() }
+                .flatMap { it.suggestProjectPaths(projectDir) + it.suggestHomePaths() }
                 .mapNotNull { RsToolchainProvider.getToolchain(it.toAbsolutePath()) }
-                .firstOrNull()
+                .let { toolchains -> toolchains.firstOrNull { "bazel" in it.presentableLocation } ?: toolchains.firstOrNull() }
+        }
+
+        fun findToolchainInBazelProject(projectRoot: File): Path? {
```

Comment:
Each Bazel project has build tools, package managers etc. declared in a `WORKSPACE` file. For a Rust project it'll include the Rust stdlib, which is downloaded to a path accessible from `{projectRoot}/bazel-$projectName`.